### PR TITLE
Add must have alt image rule and tests

### PR DESCRIPTION
**Pull request by nayres**
🎟️ [Asana Task](https://app.asana.com/0/1203331966445516/1203414786494817)

---

## Description

This PR adds a rule to ensure all content file images have alt text.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203555682638503

### CODE REVIEW

File: packages/content-conformance/src/rules/image-must-have-alt.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export default {
+  type: 'content',
+  id: 'image-must-have-alt',
+  description: 'Images must have alt text to ensure accessibility for everyone',
+  executor: {
+    async contentFile(file, context) {
+      file.visit(['image'], (node) => {
+        if (!node.alt) {
+          context.report(
+            `Warning: Found MDX image with undefined alternate text. Even if an image is decorative, it's important for alt to be set to an empty string. Please define alt text the syntax "![Some alt text.](/some-image.jpg)". Image details: ${JSON.stringify(
```
[the remark this answers is not in the document]
Comment (BRKalow) [reply packages/content-conformance/src/rules/image-must-have-alt.js:12]: +1 on omitting `warning:`. What do we think about this `The image with url {url} is missing alternate text. Providing alt text on an image is important for accessibility. Add alt text on the image with the following syntax "My alt text."`

File: packages/content-conformance/src/__tests__/cli.test.ts [hunk 2]
```diff
@@ -24,7 +24,8 @@ describe('Content-conformance CLI', () => {
     )
 
     expect(res).toMatchInlineSnapshot(`
-"Configuring content conformance runner...
+"[ 'content-check' ]
```
Comment (BRKalow) [inline packages/content-conformance/src/__tests__/cli.test.ts:27]: 👀 These snapshot updates don't look quite right, was this from a local change?

Comment (nayres) [reply packages/content-conformance/src/__tests__/cli.test.ts:27]: oh no! That was from our code sess when we were checking the default command

**Review by BRKalow — APPROVED** [review]

**Comment by BRKalow** [comment]
@nayres It looks like there are a few TS errors in the CLI:
```
Error: src/cli.ts(32,7): error TS2578: Unused '@ts-expect-error' directive.
Error: src/cli.ts(48,9): error TS2578: Unused '@ts-expect-error' directive.
```

**Pull request merged**